core/src/main/resources/lib/audio.fjs

```diff
@@ -0,0 +1,177 @@
+    playPauseC = fromEvent('click', a.playPauseButton)
+      .map((e: Event) => ((e.target: any).closest(playPauseButtonSelector): ?HTMLButtonElement))
+      .filter((e: ?HTMLButtonElement) => !!e);
+    playPauseC.tap(onPlayPause(a));
+
+    //TODO: Do we have to capture scrub events?
+    scrubC = fromEvent('click', a.scrubber)
+      .map((e: Event) => ((e.target: any).closest(progressBarSelector): ?HTMLElement))
+      .filter((e: ?HTMLElement) => !!e);
+    scrubC.tap(onScrub(a));
+
+    playerTimeUpdateC = fromEvent('timeupdate', a.player)
+      .map((e: Event) => ((e.target: any).closest(audioPlayerSelector): ?HTMLElement))
+      .filter((e: ?HTMLElement) => !!e);
+    playerTimeUpdateC.tap(onTick(a));
+
+    observer = onVisible(a);
+    viewport.observe(root, 1, observer);
+
+    return Promise.resolve();
+  };
+  
+  const stop = () => {
+    playPauseC.close();
+    scrubC.close();
+    viewport.unobserve(root, 1, observer);
+  };
+
+  const onPlayPause = (p: Audio) => (a: Action): void => {
+    record(p.audioId, a);
+    if (p.player.paused) {
+      setPlayingState(dom, p.playPauseButton);
+      p.player.play();
+      p.player.ontimeupdate = updateTime(dom, p);
+    } else {
+      setPausedState(dom, p.playPauseButton);
+      p.player.pause();
+    }
+  };
+
+  const onScrub = (p: Audio) => (a: Action): void => {
+    // record(p.audioId, a);
+    // presumably this should send an event to the audio player to skip in the direction of mouse travel?
+  };
+
+  const onTick = (p: Audio) => (a: Action): void => {
+    updateTime(dom, p);
+  };
+
+  const onVisible = (p: Audio) => (ratio: number): void => {
+    if (ratio >= 1) {
+      record(p.audioId, Actions.VIEW);
+      setupTime(dom, p);
+      viewport.unobserve(root, 1, observer);
+    }
+  };
+
+  const record = (id: string, action: Action) => {
+    ophan.record({
+      componentEvent: {
+        component: {
+          componentType,
+          id,
+          products: [],
+          labels: []
+        },
+        action
+      }
+    });
+  };
+  
+  const runTry = (): Try<Audio> => {
+    const playPauseButton = (root.querySelector(playPauseButtonSelector): ?HTMLElement);
+    const scrubber = (root.querySelector(progressSlider): ?HTMLElement);
+    const audio = (root.querySelector(audioContainerSelector): ?HTMLElement);
+    const player = (root.querySelector(audioPlayerSelector): ?HTMLElement);
+    const timePlayed = (root.querySelector(timePlayedSelector): ?HTMLElement);
+    const timeDuration = (root.querySelector(timeDurationSelector): ?HTMLElement);
+
+    return audio && player && playPauseButton && timePlayed && scrubber
+      ? Object.freeze({
+        atomId: (root.dataset.atomId: string),
+        audioId: (root.dataset.atomId: string),  //TODO: Where should this come from? Should it even be here?
+        player: player,
+        audioType: (root.dataset.audioType: string),  //TODO: Not currently available. Do we need it?
```

Comment:
I suppose that's because you used the snippet as an example, since snippets come in four flavours (guide, q&a, etc.) we had to keep track of that; here it is not useful though.